pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -293,3 +296,15 @@ func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) string
 	deviceSetName = strings.Replace(deviceSetName, ".", "-", -1)
 	return fmt.Sprintf("%s-%s-%d", deviceSetName, cleanName, setIndex)
 }
+
+func createValidImageVersionLabel(image string) string {
+	// regex to replace characters used by image name format that are not allowed in label values
+	re := regexp.MustCompile("[/:]")
+	cephImageVersion := re.ReplaceAllString(image, "_")
+
+	if validation.IsValidLabelValue(cephImageVersion) != nil {
+		logger.Infof("Value:%s contains invalid character, skipping adding label", image)
```

Comment:
nit: it's a common pattern in rook to use %q since it puts quotes around the value
```suggestion
		logger.Infof("image %q contains invalid character, skipping adding label", image)
```